back/spec/acceptance/cosponsorships_spec.rb

```diff
@@ -0,0 +1,59 @@
+
+
+require 'rails_helper'
+require 'rspec_api_documentation/dsl'
+
+
+resource 'Cosponsorships' do
+  
+  explanation 'Cosponsorships are the relationships between users and ideas. They can be pending or accepted.'
+  header 'Content-Type', 'application/json'
+
+  before do
+    header 'Content-Type', 'application/json'
+    @project = create(:single_phase_proposals_project)
+    @idea = create(:idea, project: @project, phases: @project.phases)
+    @user = create(:user)
+    @cosponsorship = create(:cosponsorship, user: @user, idea: @idea)
+  end
+
+  get 'web_api/v1/ideas/:idea_id/cosponsorships' do
+    with_options scope: :page do
+      parameter :number, 'Page number'
+      parameter :size, 'Number of cosponsorships per page'
+    end
+
+    before do
+      @cosponsorships = create_list(:cosponsorship, 2, idea: @idea)
+    end
+
+    example 'List all cosponsorships for an idea' do
+      do_request idea_id: @idea.id
+      expect(status).to eq 200
+      expect(response_data.size).to eq 3
+    end
+  end
+
+  context 'when the user is the invited user' do
```

Comment:
I can't get this spec to pass, I think I'm doing something wrong with the user authentication but I'm not sure how to go about it. I need to authenticate as the user who is associated with the cosponsorship so that I can accept the invitation.